### PR TITLE
feat(rest-api): `/destinationTx` returns status of `refunded` when RFQ transaction has been refunded [SLT-350]

### PR DESCRIPTION
**Pull request by abtestingalpha**
* Returns `refunded` status alongside response when RFQ transaction has been refunded
* Returns `404` when neither `fromInfo` or `toInfo` are found
* General refactoring and cleanup of `destinationTx` controller into `utils` and `serializers`
* Remove detailed stack errors from `500` response as they exist in logging
* Refactored graphql queries with query constructor and fetch util

## Summary by CodeRabbit

- **New Features**
	- Introduced new constants for the OmniRPC GraphQL explorer URL and a mapping for Fast Bridge Router addresses based on chain IDs.
	- Added a new utility function for fetching bridge transaction details, streamlining data retrieval processes.
	- Implemented a new function to construct GraphQL queries for bridge transactions, enhancing flexibility in data requests.

- **Bug Fixes**
	- Improved error handling by updating response structures in transaction status controllers.

- **Tests**
	- Expanded test coverage for transaction-related functionalities, ensuring robust error handling and response validation.

### CODE REVIEW

File: packages/rest-api/src/utils/constructBridgeTransactionsQuery.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          value
+          USDValue
+          tokenSymbol
+          tokenAddress
+          blockNumber
+          formattedTime
+        }
+      }
+    }
+  `
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/utils/constructBridgeTransactionsQuery.ts:45]: Not needed at moment, if we expand this query and need to modify it more then it makes sense

File: packages/rest-api/src/utils/fetchBridgeTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  if (
+    graphqlData.data &&
+    graphqlData.data.bridgeTransactions &&
+    graphqlData.data.bridgeTransactions.length > 0
+  ) {
+    return graphqlData.data.bridgeTransactions[0] as BridgeTransaction
+  }
+
+  return null
+}
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/utils/fetchBridgeTransaction.ts:37]: These errors caught in the controllers where its used

[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/utils/fetchBridgeTransaction.ts:37]: Errors are caught above

File: packages/rest-api/src/constants/index.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { CHAINS } from './chains'
 
+export const EXPLORER_GRAPHQL_URL = 'https://explorer.omnirpc.io/graphql'
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/constants/index.ts:3]: its a publicly known url

File: packages/rest-api/src/utils/fetchBridgeTransaction.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  const query = constructBridgeTransactionsQuery(params)
+
+  const graphqlResponse = await fetch(EXPLORER_GRAPHQL_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ query }),
+  })
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/utils/fetchBridgeTransaction.ts:24]: Errors are caught above

File: packages/rest-api/src/constants/index.ts [hunk 5]
```diff
@@ -27,3 +29,15 @@
...
+  '1': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '10': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '56': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '480': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '8453': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '42161': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '59144': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '81457': '0x00cD000000003f7F682BE4813200893d4e690000',
+  '534352': '0x00cD000000003f7F682BE4813200893d4e690000',
+}
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/constants/index.ts:43]: Comment is incorrect

**Review by trajan0x — APPROVED** [review]

**Pull request merged**